### PR TITLE
Feature/token gated chat use case

### PR DESCRIPTION
**Pull request by mmoloksher**
Add a Token Gated Chat Use Case
Accompanying docs can be found here... https://github.com/LiveLike/engagement-sample-app/wiki/Token-Gated-Chat-Tutorial

### CODE REVIEW

File: Sample App/Use Cases/Token Gated Chat/TokenGatedChatUseCase.swift [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                
+                // 2. If token gates enabled on the chatroom, retrieve the smart contract address
+                // NOTE: for simplicity we are only using the first token gate available
+                if let firstTokenGatedAddress = chatroomInfo.tokenGates.first {
+                    
+                    // Create an Etherscan account to receive and API key
+                    // https://docs.etherscan.io/getting-started/viewing-api-usage-statistics#creating-an-api-key
+                    let etherscan = EtherscanAPI(
+                        ethNetwork: .ropsten,
+                        apiKey: "<API KEY>"
```
Comment (mjelzon) [inline Sample App/Use Cases/Token Gated Chat/TokenGatedChatUseCase.swift:147]: Should you add a text field for the user to input an api key. The person checking out this use case may not be a developer and won't have a means to change this.

**Review by mjelzon — CHANGES_REQUESTED** [review]

File: Sample App/Use Cases/Token Gated Chat/TokenGatedChatUseCase.swift (hunk 1 above)
Comment (mmoloksher) [reply Sample App/Use Cases/Token Gated Chat/TokenGatedChatUseCase.swift:147]: Not sure if necessary. This use case goes hand in hand with the tutorial, it's meant strictly for integrators.

Comment (mjelzon) [reply Sample App/Use Cases/Token Gated Chat/TokenGatedChatUseCase.swift:147]: I'm not seeing anywhere in the docs saying this needs to be replaced with an actual token. Could be missed.

Comment (mmoloksher) [reply Sample App/Use Cases/Token Gated Chat/TokenGatedChatUseCase.swift:147]: good point, ill add to the docs

Comment (mmoloksher) [reply Sample App/Use Cases/Token Gated Chat/TokenGatedChatUseCase.swift:147]: @mjelzon  added more instructions here... 
- [Once an API key is acquired, pass it in the `EtherscanAPI` constructor in `TokenGatedChatUseCase.swift:147`](https://github.com/LiveLike/engagement-sample-app/wiki/Token-Gated-Chat-Tutorial#etherscanapi)
- https://github.com/LiveLike/engagement-sample-app/wiki/Token-Gated-Chat-Tutorial#3-query-the-blockchain

**Review by mjelzon — APPROVED** [review]

**Pull request merged**